.github/workflows/auto-create-release-pr.yml

```diff
@@ -0,0 +1,113 @@
+            url="https://api.github.com/repos/${GITHUB_REPOSITORY}/branches?per_page=100&page=${page}"
+            curl -sS -H "Authorization: token $GITHUB_TOKEN" -H "Accept: application/vnd.github.v3+json" "$url" -o "$resp"
+
+            # Append this page's branches to all_branches
+            cat "$resp" >> "$all_branches"
+
+            count=$(jq length "$resp")
+            if [ "$count" -lt 100 ]; then
+              total_pages=$page
+              break
+            fi
+            page=$((page + 1))
+          done
+          echo "Fetched $total_pages pages, total branches: $(jq -s 'map(length) | add' "$all_branches")"
+
+          # Process all branches: extract {name, semver} for matches, sort desc by semver
+          jq -s 'add | [ .[] | .name | select(test("^Version-v[0-9]+\\.[0-9]+\\.[0-9]+$") or test("^release/[0-9]+\\.[0-9]+\\.[0-9]+$")) | {name: ., semver: (if test("^Version-v") then ltrimstr("Version-v") else ltrimstr("release/") end) } ] | sort_by( .semver | split(".") | map(tonumber) ) | reverse' "$all_branches" > all_versions.json
+
+          # Print all found versions (sorted desc)
+          echo "All found versions (sorted desc): $(jq '[ .[].semver ]' all_versions.json)"
+
+          # Print all found matching branches (sorted by semver desc)
+          echo "All found matching branches:"
+          jq -r '.[].name' all_versions.json || echo "No matching branches found."
+
+          # Filter to those with semver strictly lower than current
+          jq --arg semver "$semver" '[ .[] | select( .semver as $v | $semver | split(".") as $c | $v | split(".") as $p | ($p[0] | tonumber) < ($c[0] | tonumber) or (($p[0] | tonumber) == ($c[0] | tonumber) and (($p[1] | tonumber) < ($c[1] | tonumber) or (($p[1] | tonumber) == ($c[1] | tonumber) and ($p[2] | tonumber) < ($c[2] | tonumber)))) ) ]' all_versions.json > filtered_versions.json
+
+          # Print filtered versions (still sorted desc, so [0] is highest lower)
+          echo "Filtered versions (< $semver, sorted desc): $(jq '[ .[].semver ]' filtered_versions.json)"
+
+          # Select the highest lower: first in filtered list
+          if [ "$(jq length filtered_versions.json)" -eq 0 ]; then
+            previous_ref="main"
+            echo "No versions lower than $semver found, falling back to: ${previous_ref}"
+          else
+            highest_lower="$(jq -r '.[0].semver' filtered_versions.json)"
+            previous_ref="$(jq -r '.[0].name' filtered_versions.json)"
+            echo "Selected highest version lower than $semver: ${highest_lower}"
+            echo "Selected branch: ${previous_ref}"
+            echo "Passing to previous-version-ref: ${previous_ref}"
+          fi
+          echo "previous_ref=${previous_ref}" >> "$GITHUB_OUTPUT"
+
+          # Print values passed to call-create-release-pr
+          echo "Inputs to call-create-release-pr:"
+          echo "  checkout-base-branch: main"
+          echo "  release-pr-base-branch: master"
```

Comment:
`master` can now be renamed `stable` (more context [here](https://consensys.slack.com/archives/CTQAGKY5V/p1756887981054179?thread_ts=1756824513.692929&cid=CTQAGKY5V))